#Use EB provided libz
# January 11, 2016 by B. Hajgato (Vrije Ubiversiteit Brussel)
--- Makefile.orig	2016-01-08 02:23:01.000000000 +0100
+++ Makefile	2016-01-11 16:41:25.548773718 +0100
@@ -1,6 +1,6 @@
 CC = g++
 CFLAGS = -Wall -c -I.
-COFLAGS = -Wall -O3 -ffast-math -c -I.
+COFLAGS = -march=native -Wall -O3 -ffast-math -c -I.
 PROGRAMS = rsem-extract-reference-transcripts rsem-synthesis-reference-transcripts rsem-preref rsem-parse-alignments rsem-build-read-index rsem-run-em rsem-tbam2gbam rsem-run-gibbs rsem-calculate-credibility-intervals rsem-simulate-reads rsem-bam2wig rsem-get-unique rsem-bam2readdepth rsem-sam-validator rsem-scan-for-paired-end-reads
 
 .PHONY : all ebseq clean
@@ -8,6 +8,8 @@
 all : $(PROGRAMS)
 
 sam/libbam.a :
+	sed -e 's/ -lz/ -L\$$\{EBROOTZLIB\}\/lib -lz/' -i sam/Makefile
+	sed -e 's/ -lz/ -L\$$\{EBROOTZLIB\}\/lib -lz/' -i sam/bcftools/Makefile
 	cd sam ; ${MAKE} all
 
 Transcript.h : utils.h
@@ -52,7 +54,7 @@
 
 
 rsem-parse-alignments : parseIt.o sam/libbam.a
-	$(CC) -o rsem-parse-alignments parseIt.o sam/libbam.a -lz -lpthread 
+	$(CC) -o rsem-parse-alignments parseIt.o sam/libbam.a -L${EBROOTZLIB}/lib -lz -lpthread 
 
 parseIt.o : utils.h GroupInfo.h Read.h SingleRead.h SingleReadQ.h PairedEndRead.h PairedEndReadQ.h SingleHit.h PairedEndHit.h HitContainer.h SamParser.h Transcripts.h sam/sam.h sam/bam.h parseIt.cpp
 	$(CC) -Wall -O2 -c -I. parseIt.cpp
@@ -87,7 +89,7 @@
 WriteResults.h : utils.h my_assert.h GroupInfo.h Transcript.h Transcripts.h RefSeq.h Refs.h Model.h SingleModel.h SingleQModel.h PairedEndModel.h PairedEndQModel.h
 
 rsem-run-em : EM.o sam/libbam.a
-	$(CC) -o rsem-run-em EM.o sam/libbam.a -lz -lpthread
+	$(CC) -o rsem-run-em EM.o sam/libbam.a -L${EBROOTZLIB}/lib -lz -lpthread
 
 EM.o : utils.h my_assert.h Read.h SingleRead.h SingleReadQ.h PairedEndRead.h PairedEndReadQ.h SingleHit.h PairedEndHit.h Model.h SingleModel.h SingleQModel.h PairedEndModel.h PairedEndQModel.h Refs.h GroupInfo.h HitContainer.h ReadIndex.h ReadReader.h Orientation.h LenDist.h RSPD.h QualDist.h QProfile.h NoiseQProfile.h ModelParams.h RefSeq.h RefSeqPolicy.h PolyARules.h Profile.h NoiseProfile.h Transcript.h Transcripts.h HitWrapper.h BamWriter.h sam/bam.h sam/sam.h simul.h sam_rsem_aux.h sampling.h boost/random.hpp WriteResults.h EM.cpp
 	$(CC) $(COFLAGS) EM.cpp
@@ -97,13 +99,13 @@
 BamConverter.h : utils.h my_assert.h sam/sam.h sam/bam.h sam_rsem_aux.h sam_rsem_cvt.h bc_aux.h Transcript.h Transcripts.h
 
 rsem-tbam2gbam : utils.h Transcripts.h Transcript.h bc_aux.h BamConverter.h sam/sam.h sam/bam.h sam/libbam.a sam_rsem_aux.h sam_rsem_cvt.h tbam2gbam.cpp sam/libbam.a
-	$(CC) -O3 -Wall tbam2gbam.cpp sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall tbam2gbam.cpp sam/libbam.a -L${EBROOTZLIB}/lib -lz -lpthread -o $@
 
 rsem-bam2wig : utils.h my_assert.h wiggle.h wiggle.o sam/libbam.a bam2wig.cpp
-	$(CC) -O3 -Wall bam2wig.cpp wiggle.o sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall bam2wig.cpp wiggle.o sam/libbam.a -L${EBROOTZLIB}/lib -lz -lpthread -o $@
 
 rsem-bam2readdepth : utils.h my_assert.h wiggle.h wiggle.o sam/libbam.a bam2readdepth.cpp
-	$(CC) -O3 -Wall bam2readdepth.cpp wiggle.o sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall bam2readdepth.cpp wiggle.o sam/libbam.a -L${EBROOTZLIB}/lib -lz -lpthread -o $@
 
 wiggle.o: sam/bam.h sam/sam.h wiggle.cpp wiggle.h
 	$(CC) $(COFLAGS) wiggle.cpp
@@ -131,13 +133,13 @@
 	$(CC) $(COFLAGS) calcCI.cpp
 
 rsem-get-unique : sam/bam.h sam/sam.h getUnique.cpp sam/libbam.a
-	$(CC) -O3 -Wall getUnique.cpp sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall getUnique.cpp sam/libbam.a -L${EBROOTZLIB}/lib -lz -lpthread -o $@
 
 rsem-sam-validator : sam/bam.h sam/sam.h my_assert.h samValidator.cpp sam/libbam.a
-	$(CC) -O3 -Wall samValidator.cpp sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall samValidator.cpp sam/libbam.a -L${EBROOTZLIB}/lib -lz -lpthread -o $@
 
 rsem-scan-for-paired-end-reads : sam/bam.h sam/sam.h my_assert.h scanForPairedEndReads.cpp sam/libbam.a
-	$(CC) -O3 -Wall scanForPairedEndReads.cpp sam/libbam.a -lz -lpthread -o $@
+	$(CC) -O3 -Wall scanForPairedEndReads.cpp sam/libbam.a -L${EBROOTZLIB}/lib -lz -lpthread -o $@
 
 ebseq :
 	cd EBSeq ; ${MAKE} all
